### ISSUE TITLE
Request to recover EuroSys and ATC for OS area

### ISSUE THREAD

**Issue by realstolz** [emeryberger/CSrankings#638]
Dear Emery,

I just noted that EuroSys and USENIX ATC were excluded from Operating Systems area. I would sincerely request you to reconsider it.  

**1. Only ONE conference in OS area**

After excluding EuroSys and USENIX ATC, in fact, the OS area has only ONE conference now. Because SOSP and OSDI are both 2-year each, OSDI in even years, SOSP in odd years. The OS community also widely consider OSDI and SOPS as a single conference. For example, the important award winners of SIGOPS will be announced at the SOSP or OSDI conference including [The Hall of Fame Award](https://www.sigops.org/awards/hof/), [The Mark Weiser Award](https://www.sigops.org/awards/mw/), and [The Dennis M. Ritchie Award](https://www.sigops.org/awards/dmr/). Similar examples in the community there are many, like [Systems Research: Hall of Fame ](http://www.cs.utexas.edu/~vijay/hall.html) and [The Systems "Top 50”](http://from-a-to-remzi.blogspot.com/2013/05/the-systems-top-50.html).

Given the facts above, including just ONE conference (with very few papers) of OS area may be not reasonable and cause some problems. The first mind I have is the ranking of OS area will become much unstable, and very few institutions/faculties can be included. For example, the top 10 institutions for last ten years in operating systems area. 

```
2017(39P,25I)  2016(47P,21I)  2015(30P,27I)  2014(42P,34I)  2013(30P,24I)
1.6/MIT        1.5/UMich      1.2/Chicago    2.0/MIT        2.3/MIT
0.9/Berkeley   1.1/UWisc      1.0/MIT        1.1/UW         1.0/UNSW
0.7/Columbia   1.0/UTAustin   0.7/UMAmh      0.9/UMich      1.0/UWisc
0.7/SJTU       1.0/Berkeley   0.7/Toronto    0.8/Harvard    1.0/EPFL
0.6/Toronto    0.8/UW         0.5/UMich      0.7/Berkeley   0.8/UTAustin
0.6/UTAustin   0.7/CMU        0.5/EPFL       0.7/Toronto    0.7/Stanford 
0.6/EPFL       0.6/Tsinghua   0.5/Stanford   0.7/UWisc      0.6/Berkeley
0.5/Purdue     0.6/Toronto    0.4/SJTU       0.6/ETH        0.6/CMU
0.5/VU         0.5/MIT        0.4/Technion   0.6/EPFL       0.5/UMich
0.5/UW         0.5/VTech      0.4/UCI        0.6/Chicago    0.5/UCSD

P: Papers, I: Institutions
Sum of Adj.#/Institution
```

The Top10 institutions of last five years are various, and sometimes publishing 1 paper is enough to make you into the list and even the Top5. I think that an essential goal of CSRankings is to help students determine which institution/faculty may be best for them, while the current result of OS area seems not very representative and might be misunderstood. 

I also give a brief comparison of the total number of papers in 2017 for each area of Systems. 
(a few conferences use the number in 2016)

```
#P  Area (Conferences)
171 Computer architecture (ASPLOS/56, ISCA/54, MICRO/61)
 82 Computer networks (SIGCOMM/36, NSDI/46)
296 Computer security (CCS/151, S&P/60, USENIX Security/85)
261 Databases (SIGMOD/100, VLDB/161)
311 Design automation (DAC/177, ICCAD/134)
 87 Embedded & real-time systems (EMSOFT/26, RTAS/29, RTSS/32)
108 High-performance computing (HPDC/19, ICS/28, SC/61)
 90 Mobile computing (MobiCom/35, MobiSys/34, SenSys/21)
 69 Measurement & perf. analysis (IMC/42, SIGMETRICS/27)
111 Programming languages (PLDI/47, POPL/64)
138 Software engineering (FSE/72, ICSE/66)

 39 Operating systems (SOSP/39)  
```

**2. EuroSys and USENIX ATC are good enough**

According to the description of "How were the conferences selected?” in [FAQ](http://csrankings.org/faq.html): "The conferences listed were developed in consultation with faculty across a range of institutions, including via community surveys. These are [the most impactful](https://scholar.google.com/citations?view_op=top_venues&hl=en&vq=eng) and selective conferences for each area."

Since the H5-index is widely used to assess the impact of conferences, I first list the numbers from [General Conference Rankings](http://www.guide2research.com/topconf/).

```
RK  H5 Conference
63  41 EuroSys
66  41 USENIX ATC
77  39 OSDI
-- <12 SOSP

The conferences included in System area of CSRankings with lower rank
 RK H5 Conference (Area)
 64  41 MICRO (Computer architecture)
 75  39 CAV (Logic & verification)
 77  39 OSDI (Operating systems)
123  29 ICCAD (Design automation)
130  29 HPDC (High-performance computing)
132  29 SenSys (Mobile computing)
159  26 ICS (High-performance computing)
174  25 LICS (Logic & verification)
177  25 RTAS (Embedded & real-time systems)
205  22 RTSS (Embedded & real-time systems)
257  19 EMSOFT (Embedded & real-time systems)
--- <12 SIGMETRICS (Measurement & perf. analysis)
--- <12 MobiSys (Mobile computing)
--- <12 SOSP (Operating systems)
```

In fact, I don’t think H5-index or H5-media is a good metric to compare the conferences with various of #papers, since it is very sensitive to the number of papers accepted by the conference and the research area.

According to the description of "Why so few conferences per area?” in [FAQ](http://csrankings.org/faq.html): "Only the very top conferences in each area are listed. All conferences listed must be roughly equivalent in terms of number of submissions, selectivity, and impact to avoid creating incentives to target less selective conferences.”

```
        OSDI/SOSP       EuroSys         USENIX ATC
2017    39/232/16.8%    41/201/20.4%    60/283/21.2% 
2016    47/260/18.1%    38/182/20.9%    47/248/19.0%
2015    30/181/16.6%    32/150/21.3%    47/189/24.9%
2014    42/228/18.4%    27/147/18.4%    44/196/22.4%
2013    30/157/19.1%    28/143/19.6%    33/195/16.9%

(#Papers/#Submissions/Accept-Rate)
```

I would definitely agree Eurosys and ATC are not at the same level as OSDI/SOSP. However, I still argue that the quality of EuroSys and USENIX ATC is higher than the bar of CSRankings, since all of them (EuroSys/ATC/OSDI/SOSP) follow the similar peer review procedure, two rounds of reviews (>3 reviews + >2 reviews) and in-person PC meeting.

Moreover, a slight bias among selected conferences also exits in other areas (consider the rank in [General Conference Rankings](http://www.guide2research.com/topconf/)).

```
Computer architecture: ISCA/29 ASPLOS/39  vs.  MICRO/64
High-performance computing: SC/44  vs.  HPDC/130 ICS/159 
Logic & verification: CAV/75  vs.  LICS/174
Mobile computing: MobiCom/49  vs.  SenSys/123 MobiSys/--
Measurement & perf. analysis: IMC/47  vs.  SIGMETRICS/--
```

**3. Sharing the same community**

There are 37 faculties publish more than 3 papers of OSDI/SOSP in last five years (2013-2017). I find that 62% (23/37) and 65% (24/37) of these faculties also publish at least one paper at EuroSys and USENIX ATC papers during the same period (2013-2017) respectively.

```
2013-2017            O/S  E    A
Nickolai Zeldovich   13   2    2
Ion Stoica           7    4    0    
Ding Yuan            7    0    0
Wyatt Lloyd          7    0    1
A. Arpaci-Dusseau    6    2    1
R. Arpaci-Dusseau    6    2    1
Frans Kaashoek       6    2    1
Michael Stumm        5    0    0
Scott Shenker        5    1    2
Xi Wang              5    0    0
A. Krishnamurthy     4    1    3
Junfeng Yang         4    1    1    
Emmett Witchel       4    1    1
Eddie Kohler         4    1    0
Sylvia Ratnasamy     4    1    0
Matei Zaharia        4    1    0
Aditya Akella        4    0    1
Simon Peter          4    0    0
Jason Flinn          4    0    0
Taesoo Kim           3    1    8    
Haibo Chen           3    4    3
Jinyang Li           3    1    4
Shan Lu              3    2    2
Rong Chen 0001       3    2    1
Yuanyuan Zhou        3    2    1
Timothy Roscoe       3    2    1
Bryan Parno          3    1    2
Steven Swanson       3    1    2
Michael Walfish      3    2    0
David G. Andersen    3    1    1
Adam Chlipala        3    0    1
Baris Kasikci        3    0    1
George Candea        3    0    1    
Vijay Chidambaram    3    0    1
Michael Franklin     3    0    0
Manos Kapritsos      3    0    0
Samer Al-Kiswany     3    0    0
```

Moreover, there's a level of tacit found in the important days of these conferences (SOSP/ODSI, EuroSys, USENIX ATC) to avoid the conflict among them. Usually, the notification date of the previous conference is earlier than the deadline date of the later conference, and the conference date of the previous conference will avoid overlapping with the rebuttal/notification date of the later conference. I think that including EuroSys and USENIX ATC will not create incentives to target less selective conferences.

```
                EuroSys   ATC       SOSP      EuroSys   ATC       OSDI
                2017      2017      2017      2018      2018      2018
deadline        Oct. 21   Feb. 7    Apr. 21   Oct. 27   Feb. 6    May 3
notification    Jan. 23   Apr. 21   Aug. 7    Jan. 22   Apr. 18   Jul. 24
conference      Apr. 23   Jul. 12   Oct. 28   Apr. 23   Jul. 11   Oct. 8 
```

Finally, thank you very much for putting this great resource together and maintaining it.

Best,
Rong Chen
Shanghai Jiao Tong University

**Comment by emeryberger** [comment]
This change is the result of a very recent survey of SOSP and OSDI PC members. You can see that there is a clear delineation between those two conferences and all the rest. CSrankings uses only conferences that form a top equivalence class, and you can see that these are clearly viewed as better than the rest. The community has spoken; I am however considering adding more conferences that would not be selected by default.

<img width="1400" alt="screen shot 2017-11-29 at 12 06 52 pm" src="https://user-images.githubusercontent.com/1612723/33388377-d288c8b6-d4fd-11e7-87d2-d21d630ede1a.png">

**Comment by fycus-tree** [comment]
@emeryberger @realstolz 
Again this boils down to a difference in grouping. To "this is a top tier venue in Systems", you had "agree" or "strongly agree" responses of
- SOSP 100%
- OSDI 100%
- NSDI ~90%
- ASPLOS ~85%
- Eurosys ~80% 
(an there was 0% strong disagreement with any of them)

I'd say this shows that by a large majority, all 5 of these conferences are considered top tier by the community. However, because NSDI is "networks" and ASPLOS is "arch", they get included, while Eurosys is "os" along with OSDI & SOSP, it gets excluded. Since Eurosys is more general, and covers storage, security, networking and algorithms, it seems to be held back simply by arbitrary grouping choices. The ACM SIGOPS designation includes NSDI, ASPLOS, Eurosys, etc.  (along with MobiSys &  SenSys), so that doesn't save us either. 

**My reading of your survey is that Eurosys should be included** (as it's comparable to ASPLOS and NSDI), while USENIX ATC (which has strong disagree ~= strong agree, and a total "agreement" of only ~50%) should not. The above statistics about acceptance numbers & percentages, along with publishing in both venues seems to support that it's roughly equivalent in standard metrics as well.

**Comment by emeryberger** [comment]
I am following the standard approach of scoring Likert scales (1-5). Here are the means.

ASPLOS | EuroSys | PODC | SOCC | SOSP | SYSTOR | VEE | OSDI | USENIX ATC | FAST | NSDI
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
4.22 | 3.96 | 2.82 | 2.62 | 4.9 | 2.04 | 2.34 | 4.9 | 3.28 | 3.54 | 4.38

In short, OSDI and SOSP are practically 5. The standard deviation is almost exactly 1. USENIX is 1.5 standard deviations below SOSP and OSDI. EuroSys is almost a full standard deviation below. This is a large gap. If anything, ASPLOS and NSDI would be included before the other conferences.

**Comment by fycus-tree** [comment]
I thought ASPLOS (Architecture) and NSDI (Networks) are already included? Most 1-5 scale analysis I've seen groups "agree" and "disagree" statements into a binary decision variable. This dichotomizing is [fairly common, known as binning, collapsing, etc.](https://www.st-andrews.ac.uk/media/capod/students/mathssupport/OrdinalexampleR.pdf). 

As for Likert scales in general, there's the classic ["Likert scales: how to (ab)use them"](http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.474.608&rep=rep1&type=pdf) which emphasizes issues with using means and variances on them. I think metrics like [consensus](http://www.sciencedirect.com/science/article/pii/S0888613X06001186) make more sense. 

Partially this is because your phrasing of "is this top tier" instead of "what tier is this", it's not clear you'd get a decent normal distribution for these LIkert scales, which would weaken their use (e.g [Likert 1932 pg 21](https://legacy.voteview.com/pdf/Likert_1932.pdf)).

**Comment by emeryberger** [comment]
Yes, ASPLOS and NSDI are currently included in other categories.

All CSrankings cares about is what constitutes the top equivalence class.

It's clear that there is a sizable gap between OSDI/SOSP and the others. OSDI/SOSP form a single equivalence class. EuroSys is clearly separate. USENIX is even further away.

If SIGOPS and EuroSys decide that in fact EuroSys is co-equal with OSDI/SOSP — incidentally, a view not shared by the current chair of EuroSys! — then I would be more than happy with their decision. As it stands, I view the survey results as definitive for now.

**Comment by tongping** [comment]
I think that everyone will agree that OSDI/SOSP is much better than other conferences, even   conferences in other fields. EuroSys is certainly not as good as these two conferences. However, there are some problems that I can think of. 

Interview results may create a bias since the interviewers are from OSDI/SOSP committee members. If you asked them opinions about other system-related conferences, such as ICS, SC or HPDC, they may think that they are not as good as EuroSys as well.  Thus, we cannot adjudge these conferences by simply from which SIG group that they are in. If EuroSys is not supported by SIGOPS, then it is possibly considered to be the top conference and will be a better choice for people. Another reason is that there are much fewer papers in the OS field each year, comparing to other fields like security and SE. However, the people in this field is certainly not less than the other two groups.

I have other suggestions to rank conferences. First, we could simply get impact factor and rank them by the factor. Second, we could get more opinions from a larger community. For instance, we could ask them to rank all systems conferences. Then we could disrupt the siggroups and rank them by a larger field.

**Comment by emeryberger** [comment]
The bias you suggest does not appear to be a problem across communities. I have surveyed a number of communities by surveying the PC members from conferences that are widely acknowledged to be at least one of the top conferences in each area. For example, I surveyed just one conference's PC for both architecture (ISCA) and software engineering (ICSE), but they reached consensus on several conferences as being in a single top equivalence class. I think OSDI and SOSP PC members have considerable experience with many systems conferences; the overlap of OSDI/SOSP and EuroSys PC members is high. A consensus view from the relevant community is far more reliable and frankly more meaningful than using bibliometrics. Changing the CSmetrics model to include conferences that are not in the same top equivalence class (at least as default choices) is just not in the cards.

**Comment by fycus-tree** [comment]
@emeryberger but the equivalence class problem is only if you consider Eurosys an "os" conference. This happens in the SIGOPS's historical organization. But it's a more general systems conference, with networking, distributed systems, etc. I think many people would consider it higher tier than any of the conferences in the "mobile computing" category. 

I agree that it doesn't belong in the "os" class, but I guess I'm wondering how conferences that are... between the lines of various ACM designations. But I guess that discussion is mostly in #562

**Comment by realstolz** [comment]
@emeryberger
I think the inter-area equivalence is also important for CSRankings, especially for a general conference like EuroSys as mentioned by @fycus-tree. For example, excluding EuroSys from CSRankings would incentivize authors to submit their papers to the conferences in the other areas. 

In fact, the largest concern in my proposal is that CSRanking only includes ONE conference (SOSP/OSDI) in the OS area. Worse even, SOSP/OSDI only accept very few papers but from many areas per year.


*SOSP 2017 (39 papers): bug hunting, scalability, in-network computing, resource management, kernels, verification, potpourri, adaptation and repair, privacy, storage systems, security, understanding failures, data analytics.*

The excellence of SOSP/OSDI may limit and impair the OS area in CSRankings. Imagine having no SOSP/OSDI: OS areas would include EuroSys, USENIX ATC, and FAST.   


Another proposal is moving ASPLOS to the OS area and leaving space to the HPCA. I know there is another survey in #288. However, it may not answer the question where to place a top-tier conference (ASPLOS) if more than two areas prefer it. In fact, ASPLOS is also a general conference. 

*ASPLSO 2017: memory, verification and testing, distributed systems, compiler optimizations, storage, IoT and applications, virtualization, accelerators, security, cloud, debugging, data center architectures & power management*

**Comment by emeryberger** [comment]
I will be revisiting this decision (consistently across all communities) soon. Stay tuned.
(In the interim, I am restoring EuroSys to the rankings.)

**Comment by dants** [comment]
1) I participated in your survey -- had I known you are going to (ab)use it in this way; I wouldn't have.

2) SOSP/OSDI (=SOSPDI) is a meta-conference: a PL conference, a system security conference, a parallel-and-dist conference, a networking conference, a storage conference, a measuring and analysis conference, and probably a few more . Many people I know in those latter communities favor submitting to SOSPDI when they think they have something worthy enough, otherwise "resorting" to the top-tier conferences in their respective communities. Many others from the PL/networking/security/et al communities perceive SOSPDI as simply another top-tier conference they can submit their work to. (I believe you meet this description.) Consequently, SOSPDI should either be viewed as the "zero-tier" conference of all of "systems", or simply as a top-tier conference in MOST of the sub-categories of "systems". Either way it's special. There's really nothing like it. 

3) Under these circumstances, leaving SIGOPS with only this 0-tier ~30-papers-per-year conference is highly problematic. And not just because of the above. There are quite a few other clear reasons. For example, one could argue that ATC/Eurosys are every bit as good as, say, ASPLOS/ISCA/MICRO and are superior to, say, ICS/SC/HPDC. In particular, as a PC member of ATC and ASPLOS in the last few years, claiming that one is better than the other isn't serious IMO, to say the least; they are about the same. (Same goes for Eurosys vs. ATC, as I speculate you'll find if you survey enough of the "right" people, namely, those who participated in both PCs, rather than just in the SOSPDI PC.)  

4) I would have made an effort to specify some of the other reasons why having only SOSPDI under "OS" is wrong, but that would likely be counterproductive since you seem quite resolved, especially given you now have the statistics that support your old SOSPDI-only preference for SIGOPS. That said, it should be noted that despite these statistics, it's nevertheless largely your personal decision, because in contrast to what you claim above, the community has NOT spoken. Is there a real difference between the outcome of (i) the csrankings maintainer making, by himself, the decision regarding which OS conferences will populate the csrankings/OS subcategory, and (ii) the csrankings maintainer---rather than SIGOPS+Usenix members---determining the methodology of the survey that chooses said conferences? (Where methodology includes deciding who gets to vote as well as to define the questions being asked in the survey---others have already noted above that different questions would likely yield different results.)

5) In an effort to be productive, here's a wild idea. Perhaps there should be a top-tier "meta confs" category under "systems", consisting of (i) SOSPDI, (ii) ASPLOS, and (iii) SIGMETRICS (does IMC really play in the same league as SIGMETRICS?). Then, "OS" will get to be something sane like Eurosys/ATC/FAST, making it similar to many of the other subcategories in terms of level and number of papers. As a bonus, "architecture" will get to be ISCA/MICRO/HPCA, which also seems like the right thing to do in terms of reflecting the perception of architecture people.

Bottom line: as things stand, although I'm quite sure csrankings will do fine without me, since its definition of my community makes no sense to me, I'm no longer its user. (Including ATC/Eurosys but excluding them by default will not change that.)

**Comment by emeryberger** [comment]
To be clear, I would love for SIGOPS and USENIX to present me with a gold standard list of conferences. In its absence, I am managing things as best I can in a uniform way across communities. (As I note in my updated comment above, I am restoring Eurosys for the interim.)

**Comment by emeryberger** [comment]
On a minor note: the question asked in the survey is exactly the one I am using for all communities, specifically to ensure consistency. My aim is to treat areas equally and using the same criteria. Obviously, this is challenging for many reasons. (Creating a new meta-area is an intriguing suggestion but clearly also tricky.)

**Comment by dants** [comment]
Having a meta-conference like SOSPDI coming at the expense of "OS" (even though it's actually also a PL/networking/security/measurement/etc conference) already distorts "OS" as it is. To be fair and equivalent to the other categories, "OS" should really consistent of Eurosys/ATC/FAST, whereas SOSPDI should have its own category. Thus, to say that resurrecting just Eurosys is insufficient, would be a serious understatement. So I'm still not your user, because, as it is, csrankings is really bad for my community.

**Comment by dants** [comment]
PS: I do understand, respect, and appreciate your attempt to treat areas equally. But as a recurring PC member of ATC and Eurosys, I nevertheless find it strange and unreasonable to include one and not the other. My subjective sense is that the quality of ATC is typically a bit higher than that of Eurosys. I speculate that the small group of people you surveyed are more a part of the Eurosys community than the ATC community, which may explain their slight preference of Eurosys, on average. (Personally, I believe I ranked the two similarly in your survery, since I feel they are indistinguishable when using a discrete scale of 0--5). Regardless, the difference between the two is small enough to make the decision to include just one arbitrary and inappropriate---a negative side-effect of making the mistake of including SOSPDI in OS despite its much broader nature.

**Comment by fycus-tree** [comment]
@dants Can you propose a methodological change that would alleviate your concerns? I understand that there are issues with Emery's policy of, "follow ACM designations, conduct polls of PC members, & use the top equivalence class." However, I respect his attempt to have a consistent policy across fields and not just cherry pick "corrections" at a whim. The issue here is bound to repeat itself (for example, Emery decides to include Nature in CompBio), and so a larger methodological suggestion would be great. 

For example, we could look at different ways of grouping computer science subfields (the [HCI](https://github.com/emeryberger/CSrankings/issues/147#issuecomment-332282987) and [security](https://github.com/emeryberger/CSrankings/issues/263#issuecomment-348560641) communities have expressed concerns that they should be split out into subfields), looking at the 3 conference limit per field, looking at having more than one equivalence class, etc. We could propose that Emery follows the "mode" of his surveys, which would give Eurosys & ATC admittance. 

I think it's a little petty to say "oh well this methodology is imperfect for my field so I'm going to dismiss the entire effort". Even in my own case, CSRankings doesn't list conferences that I submit to & have published in and (as highlighted in #562) it considers publications in my fields less "important" than other fields. As the saying goes, **All models are wrong but some are useful** . I think the goal is to make a fair and useful model, not a perfect one.

**Comment by dants** [comment]
@fycus-tree 

1. Your "petty" argument can be very easily flipped around and applied to excluding conferences where one doesn't publish. It's inappropriate IMO, and I will ignore it.

2. I and others made several suggestions, here and offline (via email) regarding the methodology; I no longer wish to do that, as I feel it's counterproductive. I'm at a point where I feel comfortable disregarding the methodology and constraining myself to providing feedback regarding only the bottom-line outcome, which makes no sense to me for the reasons I outlined above, which are anything but petty. (It seems my approach is at least partially productive, as you can see above.)

3. "All models are wrong but some are useful" -- Completely agree.  "OS" once consisted of only SOSPDI. Then, after a long email discussion, it was changed to SOSPDI/Eurosys/ATC. But recently it was changed back to SOSPDI only. And then this thread happened and it's back to SOSPDI/Eurosys. Are any of these combinations more "scientific" or "consistent" than the others? Personally, I don't think so. I respect you and any one else who thinks differently, but see #2 regarding bottom line outcome. 

4. At a very fundamental level, there is no substitute for the csrankings maintainer exercising his best judgment and making a call. Which he does. I constrain myself to discussing that.
